### DISCUSSION TITLE
nodes of the subgraph that Neighborloader returns

### DISCUSSION THREAD

**Discussion by Peien429** [pyg-team/pytorch_geometric#6658] [Q&A]
Hi, I have a question about the code in the file "examples/reddit.py". 

First , the following code creates a Neighborloader.

`train_loader = NeighborLoader(data, input_nodes=data.train_mask,num_neighbors=[25, 10], shuffle=True, **kwargs)`

Second, the train_loader performs neighbor sampling and returns a subgraph.

`for batch in train_loader`

Does the 'batch(subgraph)' only contains the nodes of train set? I find that there are nodes of test set in the subgraph. So why does the following code compute loss for training without distinguishing the nodes of test set?

`for batch in train_loader:`
        `optimizer.zero_grad()`
        `y = batch.y[:batch.batch_size]`
        `y_hat = model(batch.x, batch.edge_index.to(device))[:batch.batch_size]`
        `loss = F.cross_entropy(y_hat, y)`

**Comment by LukeLIN-web** [comment]
Does the 'batch(subgraph)' only contains the nodes of the train set? 
No, it contains nodes of the test set. 
Because `[:batch.batch_size]` will extract the y hat of the nodes of the train set

**Reply by Peien429** [reply]
Hi @LukeLIN-web, I did a test and found that first batch_size nodes in the batch were all training set nodes.
Thanks a lot for your reply.
